### PR TITLE
optimize the file copy strategy of `goc build`

### LINKED ISSUE #126: File copy strategy of `goc build` needs to be optimized
Some files like `unix socket` is not allowed to be copied. When project files contians these file, `goc build .` will return an error. The file copy strategy needs optimize.

### PR DESCRIPTION
**Pull request by tongjingran**
fix #126 and #144

### CODE REVIEW

File: pkg/build/tmpfolder.go [hunk 1]
```diff
@@ -114,9 +107,7 @@ func (b *Build) mvProjectsToTmp() error {
 		}
 	} else if b.IsMod == false && b.Root == "" {
 		b.TmpWorkingDir = b.TmpDir
-		b.cpNonStandardLegacy()
-	} else {
-		return fmt.Errorf("unknown project type: %w", ErrShouldNotReached)
```
Comment (tongjingran) [inline pkg/build/tmpfolder.go:119]: code will never be reached

**Comment by lyyyuna** [comment]
choose another name for `cpLegacyProject `

**Comment by tongjingran** [comment]
> choose another name for `cpLegacyProject `

change cpLegcyProject to cpProject

**Comment by lyyyuna** [comment]
In this PR, we no longer consider project under Go 1.11.0-1.11.4, which allows `vendor` directory place in the `GOPATH` directly.

In Qiniu, we used to have some projects can only be compiled before Go 1.11.4 (wrong vendor location), so `goc` has lots logic to be compatible with this.

Now we decide to remove this support for such structure. If you have similar issues, you should consider to  upgrade your Go toolchain to fix this.

**Pull request merged**